lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1355,6 +1355,44 @@ def compute_timestamp(resource_type, record, time)
         ts_secs = timestamp.tv_sec
         ts_nanos = timestamp.tv_nsec
       end
+      ts_secs = begin
+                  Integer ts_secs
+                rescue ArgumentError, TypeError
+                  ts_secs
+                end
+      ts_nanos = begin
+                   Integer ts_nanos
+                 rescue ArgumentError, TypeError
+                   ts_nanos
+                 end
+
+      # Adjust timestamps from the future.
+      # There are two cases:
+      # 1. The parsed timestamp is later in the current year:
+      # This can happen when system log lines from previous years are missing
+      # the year, so the date parser assumes the current year.
+      # We treat these lines as coming from last year.  This could label
+      # 2-year-old logs incorrectly, but this probably isn't super important.
+      #
+      # 2. The parsed timestamp is past the end of the current year:
+      # Since the year is different from the current year, this isn't the
+      # missing year in system logs.  It is unlikely that users explicitly
+      # write logs at a future date.  This could result from an unsynchronized
+      # clock on a VM, or some random value being parsed as the timestamp.
+      # We reset the timestamp on those lines to the default value and let the
+      # downstream API handle it.
+      if timestamp
+        next_year = Time.mktime(current_time.year + 1)
+        one_day_later = current_time.to_datetime.next_day.to_time
+        if timestamp >= next_year # Case 2.
+          ts_secs = 0
+          ts_nanos = 0
+        elsif timestamp >= one_day_later # Case 1.
+          adjusted_timestamp = timestamp.to_datetime.prev_year.to_time
+          # The value of ts_nanos should not change when subtracting a year.
```

Comment:
Let's move this below the `ts_secs` assignment.

Reply:
Done.